src/lib/mina_ledger/ledger.ml

```diff
@@ -201,6 +277,53 @@ module Ledger_inner = struct
     let _base, mask = create_ephemeral_with_base ~depth () in
     mask
 
+  type converting_config =
+    { primary_directory_name : string option
+    ; converting_directory_name : string option
+    }
+
+  let default_converting_directory_name primary_directory_name =
+    primary_directory_name ^ "_converting"
+
+  let converting_directory_name ~cfg ~primary_directory =
+    Option.first_some cfg.converting_directory_name
+    @@ Option.map primary_directory ~f:default_converting_directory_name
+
+  let empty_converting_config : converting_config =
+    { primary_directory_name = None; converting_directory_name = None }
+
+  let create_converting ?(cfg = empty_converting_config) ~depth () =
+    let open Async in
+    let db1 = Db.create ?directory_name:cfg.primary_directory_name ~depth () in
+    let db2 =
+      let directory_name =
+        converting_directory_name ~cfg ~primary_directory:(Db.get_directory db1)
+      in
+      Unstable_db.create ?directory_name ~depth ()
+    in
+    let%map.Deferred.Or_error converting_ledger =
+      let open Deferred.Let_syntax in
+      if Unstable_db.num_accounts db2 = 0 then
+        Deferred.Or_error.return
+        @@ Converting_ledger.create_with_migration db1 db2
+      else
+        let%bind db1_accounts = Db.accounts db1 in
+        let%bind db2_accounts = Unstable_db.accounts db2 in
+        (* TODO: check if relying on identifier is enough for equality *)
```

Comment:
noted

Reply:
You could do something similar to what's done in the merkle ledger tests for database equality:

https://github.com/MinaProtocol/mina/blob/76b89993962c300bbe90e032dc9f8198160812bd/src/lib/merkle_ledger_tests/test_converting.ml#L194-L199

Synced ledgers should have accounts at identical indices.

Reply:
Because of https://github.com/MinaProtocol/mina/pull/17428#discussion_r2198277110 I don't think we need to worry about the speed of account conversion. Hopefully.